Source/Dafny/DafnyPipeline.csproj

```diff
@@ -34,4 +34,28 @@
   <ItemGroup>
     <Content Include="DafnyPrelude.bpl" CopyToOutputDirectory="PreserveNewest" />
   </ItemGroup>
+
+  <ItemGroup>
+    <EmbeddedResource Include="..\DafnyRuntime\DafnyRuntime.csproj">
+      <LogicalName>DafnyRuntime.csproj</LogicalName>
+    </EmbeddedResource>
```

Comment:
Would this also work if done in `DafnyRuntime`?

Reply:
Would adding that dependency be so bad?

Reply:
It seems like it would be sort of a strange thing to do, to me. The `DafnyRuntime` project is a C# library intended for use by generated C# code. It doesn't make sense to me for it include, for instance, `DafnyRuntime.js` as an embedded resource. Said differently, it makes more sense, in my view, for the compiler to bake in the code for the various runtime systems, rather than having the runtime system for one language include the runtime systems for other languages.

Reply:
I disagree, `DafnyRuntime.csproj` already includes the runtimes for all target languages and is included by `Dafny.sln`. In fact, we probably shouldn't embed it here at all.

Reply:
I think we should either get rid of the `csproj` entirely, or use it as much as possible.

Reply:
Another way to look at it: if we embed those files in `DafnyRuntime.csproj`, then the `DafnyRuntime.dll` file that users download from NuGet to link with their generated C# code will also include the text of the various other language runtimes. It wouldn't cause problems to do that, but it seems awkward to me.

Reply:
I think we should have separate projects for each back-end, and each of those should include their own runtime, and then `DafnyDriver.csproj` should reference those, not DafnyPipeline.

Without making all those changes, I think `DafnyDriver.csproj` can reference `DafnyRuntime.csproj`

Reply:
Yeah, I like the main proposal you're making, @keyboardDrummer. Each back-end should be a separate project.

Given the current state of the code, though, the reason for embedding those resources in `DafnyPipeline` for the moment is so that the compilers can get a reference to them through [`GetExecutingAssembly`](https://docs.microsoft.com/en-us/dotnet/api/system.reflection.assembly.getexecutingassembly?view=net-6.0). I originally put them in `DafnyDriver`. It would be possible to move them back into `DafnyDriver` and instead use [`GetEntryAssembly`](https://docs.microsoft.com/en-us/dotnet/api/system.reflection.assembly.getentryassembly?view=net-6.0).

I had previously only seen methods in the `Assembly` class to load by type, and types in `DafnyRuntime` aren't available in the compilers, which are currently in `DafnyPipeline`, unless we make `DafnyRuntime` a dependency of `DafnyPipeline`, which is pretty much the opposite of what you're proposing. We could potentially use [`Load`](https://docs.microsoft.com/en-us/dotnet/api/system.reflection.assembly.load?view=net-6.0) to load `DafnyRuntime` by name. I prefer something that doesn't require a name as a string, but not strongly.